app/models/user_ability.rb

```diff
@@ -6,12 +6,15 @@ def initialize(user)
 
     can :manage, :all if user.penning?
     can :create, Request, creditor_id: user.id
-    can [:confirm, :decline], Request, debtor_id: user.id
+    can [:confirm], Request do |r|
+      (r.debtor_id == user.id) && (user.balance - r.amount >= 0)
```

Comment:
Could be `... >= Rails.application.config.minimum_amount` to allow for more flexibility in the future